src/test/scala/is/hail/methods/PCRelateSuite.scala

```diff
@@ -0,0 +1,240 @@
+import is.hail.distributedmatrix.DistributedMatrix.implicits._
+
+class PCRelateSuite extends SparkSuite {
+  private val blockSize: Int = 8192
+
+  private def toI(a: Any): Int =
+    a.asInstanceOf[Int]
+
+  private def toD(a: Any): Double =
+    a.asInstanceOf[Double]
+
+  private def toS(a: Any): String =
+    a.asInstanceOf[String]
+
+  def runPcRelateHail(vds: VariantDataset, pcs: DenseMatrix, maf: Double): Map[(String, String), (Double, Double, Double, Double)] =
+    PCRelate.toPairRdd(vds, pcs, maf, blockSize).collect().toMap.asInstanceOf[Map[(String, String), (Double, Double, Double, Double)]]
+
+  def runPcRelateR(
+    vds: VariantDataset,
+    maf: Double,
+    rFile: String = "src/test/resources/is/hail/methods/runPcRelate.R"): Map[(String, String), (Double, Double, Double, Double)] = {
+
+    val tmpfile = tmpDir.createTempFile(prefix = "pcrelate")
+    val localTmpfile = tmpDir.createLocalTempFile(prefix = "pcrelate")
+    val pcRelateScript = tmpDir.createLocalTempFile(prefix = "pcrelateScript")
+
+    vds.exportPlink(tmpfile)
+
+    for (suffix <- Seq(".bed", ".bim", ".fam")) {
+      hadoopConf.copy(tmpfile + suffix, localTmpfile + suffix)
+    }
+
+    s"Rscript $rFile ${uriPath(localTmpfile)} ${maf}" !
+
+    val columns = Map(
+      ("ID1", TString),
+      ("ID2", TString),
+      ("nsnp", TDouble),
+      ("kin", TDouble),
+      ("k0", TDouble),
+      ("k1", TDouble),
+      ("k2", TDouble))
+    val separator = " +"
+
+    hadoopConf.copy(localTmpfile + ".out", tmpfile + ".out")
+
+    val (_, rdd) = TextTableReader.read(sc)(Array(tmpfile + ".out"), columns, separator=separator)
+    rdd.collect()
+      .map(_.value)
+      .map { ann =>
+      val row = ann.asInstanceOf[Row]
+      val id1 = toS(row(0))
+      val id2 = toS(row(1))
+      val nsnp = toD(row(2)).toInt
+      val kin = toD(row(3))
+      val k0 = toD(row(4))
+      val k1 = toD(row(5))
+      val k2 = toD(row(6))
+      ((id1, id2), (kin, k0, k1, k2))
+    }
+      .toMap
+  }
+
+  def compareDoubleQuartuplets(cmp: (Double, Double) => Boolean)(x: (Double, Double, Double, Double), y: (Double, Double, Double, Double)): Boolean =
+    cmp(x._1, y._1) && cmp(x._2, y._2) && cmp(x._3, y._3) && cmp(x._4, y._4)
+
+  @Test
+  def trivialReference() {
+    val genotypeMatrix = new BDM(4,8,Array(0,0,0,0, 0,0,1,0, 0,1,0,1, 0,1,1,1, 1,0,0,0, 1,0,1,0, 1,1,0,1, 1,1,1,1)) // column-major, columns == variants
+    val vds = vdsFromGtMatrix(hc)(genotypeMatrix, Some(Array("s1","s2","s3","s4")))
+    val pcsArray = Array(0.0, 1.0, 1.0, 0.0,  1.0, 1.0, 0.0, 0.0) // NB: this **MUST** be the same as the PCs used by the R script
+    val pcs = new DenseMatrix(4,2,pcsArray)
+    val us = runPcRelateHail(vds, pcs, maf=0.0)
+    val truth = PCRelateReferenceImplementation(vds, pcs, maf=0.0)
+    assert(mapSameElements(us, truth, compareDoubleQuartuplets((x, y) => math.abs(x - y) < 1e-14)))
+  }
+
+  // @Test
+  def trivialReferenceMatchesR() {
```

Comment:
As discussed previously, I can't get my implementation to match R. I need to dig into the R implementation more to figure out why. I've commented out all the comparisons to the R implementation.